Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6640.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms.Controls.Issues
+{
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 6640, "[Android] Crash when app go background", PlatformAffected.Android)]
+	public class Issue6640 : TestContentPage
```

Comment:
Can you add a UITest for this?

Reply:
See my comment : https://github.com/xamarin/Xamarin.Forms/pull/6715#issuecomment-507080479

Reply:
Now that unhook is part of dispose (https://github.com/xamarin/Xamarin.Forms/pull/6715/files#diff-edc32d3b021f02879f303175242f36b7R174) instead of DestroyView is there a way to emulate this behavior without the backgrounding? Perhaps after the mainpage is swapped out https://github.com/xamarin/Xamarin.Forms/pull/6715/files#diff-fbd24038d82294fdd7a696c78d6e02d4R59 the item could be removed from the collection to trigger the dispose? 

```
Application.Current.MainPage = new ContentPage
														{
															Content = new StackLayout
															{
																BackgroundColor = Color.White,
																Children =
																{
																	new Label
																	{
																		Text = $"Press Back Arrow to send application to background and wait few seconds.{Environment.NewLine}Application must not crash.",
																		AutomationId = "ReadyToGoBack"
																	}
																}
															}
														};

														Device.BeginInvokeOnMainThread(() =>
														{
															this.CreateContentPage();
															this.Items.RemoveAt(0);
														});
```

or something similar to that? We just need to trigger the destroy or dispose in order to have a UI test for this right? because that's all the backgrounding is really doing

Reply:
I think the only way to trigger the bug is on Activity Destroy.
Because it involve two methods :
- ShellRenderer.Dispose : Called when the main page change for exemple
- ShellSectionRenderer.OnDestroy : Called .... when Android want, and from this [stackoverflow answer](https://stackoverflow.com/a/17195813) it's only garanted when the fragment activity is also destroyed. This statement is coherent with the issue reproduction which is only triggered on activity destroy.
It's for that reason that I have replaced all the OnDestroy with Dispose.